clang/lib/CIR/CodeGen/CIRGenStmt.cpp

```diff
@@ -303,8 +303,7 @@ mlir::LogicalResult CIRGenFunction::buildSimpleStmt(const Stmt *S,
 
   case Stmt::CaseStmtClass:
   case Stmt::DefaultStmtClass:
-    assert(0 &&
-           "Should not get here, currently handled directly from SwitchStmt");
+    return buildSwitchCase(cast<SwitchCase>(*S));
```

Comment:
I don't think you should be handling `Stmt::DefaultStmtClass` here too, right?

Reply:
We need to handle it. I added a similar test case:

```
extern "C" void default_follow_label(int v) {
  switch (v) {
    case 1:
    case 2:
      action1();
      break;
    label:
    default:
      action2();
      goto label;
  }
}
```